### PR TITLE
devops: add GitHub action workflow for PRs

### PR DESCRIPTION
**Pull request by kylejb**
## Description

Create first [GitHub Actions](https://docs.github.com/en/actions) workflow, which will execute  tests for every PR attempting to merge to `main`. Tests must passed in order to be allowed to merge.

## TODOs

Please ensure all of these TODOs are completed before asking for a review.

- [x] Ensure the branch is named correctly with the issue number. e.g: `feat/310-new-endpoint` or `bug/322-fix-missing-param`.
- [x] Update the docs.
- [x] Keep the changes backward compatible where possible.
- [x] Run the pre-commit checks successfully.
- [x] Run the relevant tests successfully.

### CODE REVIEW

**Review by Grieze — APPROVED** [review]
What's the point of github workflows and the yml files? I have no idea what they're for. I'm assuming this is for making working with github easier.

**Comment by kylejb** [comment]
This is still a work-in-progress. I want to also create a service process to run a postgresql database in order to run tests on for github actions.

> What's the point of github workflows and the yml files? I have no idea what they're for. I'm assuming this is for making working with github easier.

The yml files is what GitHub uses to create workflows in our repository. It's particularly useful for automating [CI/CD processes](https://www.redhat.com/en/topics/devops/what-is-ci-cd). I'd like to incorporate this for our pull requests.

As an example:

- We create a new Pull Request
- Workflow starts:
    - installs repository on a new server
    - runs tests on that new server (to mimic production environment)

This will help us catch critical errors earlier on. And it also will help us review each other's PRs; you won't have to run the tests locally yourself.

File: server/services/buoy/seed_buoys.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def parse_activestations(stations):
         for station in stations:
             station = station.attrs
             lon, lat = float(station.get("lon", 0.0)), float(station.get("lat", 0.0))
-            geo = f"Point({lon} {lat})"
+            geo = f"POINT({lon} {lat})"
```
Comment (Grieze) [inline server/services/buoy/seed_buoys.py:42]: Lol why? This was unnecessary. Was it OCD?

**Review by Grieze — APPROVED** [review]

File: server/services/buoy/seed_buoys.py (hunk 1 above)
Comment (kylejb) [reply server/services/buoy/seed_buoys.py:42]: I cannot not conform nor deny... 👀

**Pull request merged**